components/button-dialog.tsx

```diff
@@ -1,12 +1,13 @@
 import { kebabCase } from "lodash";
-import { ComponentPropsWithoutRef, ReactNode, ReactNodeArray } from "react";
+import { ComponentPropsWithoutRef, ReactElement, useState } from "react";
 import { Modal } from "react-bootstrap";
 
 interface Props extends ComponentPropsWithoutRef<"button"> {
   title: string;
-  footer?: ReactNode;
-  show: boolean;
-  onHide?: () => void;
+  footer?:
```

Comment:
`footer?: |` is a mistype, `footer?: ReactElement | (({ hideModal }: { hideModal: () => void }) => ReactElement` is what you're looking for

Reply:
it is not 😆! And it comes by automatic formatting!
```jsx
footer?: ReactElement | (({ hideModal }: { hideModal: () => void }) => ReactElement;
``` 
is the same as
```jsx
footer?:
 | ReactElement
 | (({ hideModal }: { hideModal: () => void }) => ReactElement;
```